src/smtsolvers/SplitData.h

```diff
@@ -0,0 +1,52 @@
+//
+// Created by prova on 31.03.21.
+//
+
+#ifndef OPENSMT_SPLITDATA_H
+#define OPENSMT_SPLITDATA_H
+
+#include "SolverTypes.h"
+#include "THandler.h"
+
+// -----------------------------------------------------------------------------------------
+// The splits
+//
+class SplitData
+{
+    bool                   no_instance;    // Does SplitData store the instance?
+
+    std::vector<vec<Lit>>  constraints;    // The split constraints
+    std::vector<vec<Lit>>  learnts;        // The learnt clauses
+
+    static char* litToString(Lit);
+    template<class C> char* clauseToString(const C&);
+    char* clauseToString(const vec<Lit>&);
+    static int getLitSize(Lit l);
+    static std::vector<vec<PtAsgn>> toPTRefs(std::vector<vec<Lit>> const & in, const THandler &thandler);
+
+public:
+    SplitData(bool no_instance = true)
+            : no_instance(no_instance)
+    { assert(no_instance); }
```

Comment:
This looks like a relic from the olden times.  I don't think we want to reproduce anymore the instance with the split, so it should be safe to remove the argument from the constructor (and then the whole constructor).

Reply:
Sure